astropy/table/tests/test_column.py

```diff
@@ -1145,3 +1145,17 @@ def test_masked_unit_conversion():
     c = table.MaskedColumn([3.5, 2.4, 1.7], name="test", unit=u.km)
     c.convert_unit_to(u.m)
     assert c.unit == (c * 2.0).unit
+
+
+@pytest.mark.parametrize("copy", [False, True])
+def test_zero_length_strings(Column, copy):
+    # Easiest way to get a zero-sized byte string is with a structured dtype.
+    data = np.array([("", 12)], dtype=[("a", "S"), ("b", "i4")])
+    col = Column(data["a"], name="a", copy=copy)
+    if copy:
+        # This is not correct, but due to a numpy bug.  Putting the assert
+        # in so we get a warning if it is fixed.
+        assert col.dtype.itemsize == 1
+        pytest.xfail("See https://github.com/numpy/numpy/issues/27301")
```

Comment:
If you xfail here, you don't get to check `col.dtype == data.dtype["a"]`, is that intentional?

Also xfail like this does not benefit from our `xfail_strict = True` setting. That is, if this bug is fixed, it would fail in assert anyway instead of xpass-ing. So I don't see the point of adding xfail here.

Reply:
@pllim - yes, sorry, this is rather hacky, but I think if I do `pytest.xfail` in the routine, execution stops immediately, i.e., the following `assert` are not executed anymore. Basically, I need an `xfail` on top just for `copy=True`... do you know how to do that?

Reply:
Alas, `pytest.xfail` (unlike `pytest.mark.xfail`) does not take a condition.

Ah, okay, so you really want it to be 1 but it is not currently, so without the xfail, it would fail for real?

Reply:
No, it is `1` currently, which is a bug in numpy, so if they fix it and it becomes `0`, we get a failure and will realize we can change this to a `if NUMPY_LT_????`... 

I probably should add a better comment!!

Reply:
Or shall I just make two test functions, `xfail`'ing the second?

Reply:
Personally I think it is more readable if we do not parametrize this test function, but rather separate out the xfail-ing case and use `pytest.mark.xfail` decorator. I think that way, we can rely on `xfail_strict = True` and find out when numpy ever fixes the problem.

Reply:
https://docs.pytest.org/en/7.0.x/how-to/skipping.html#skip-xfail-with-parametrize

Reply:
@mhvk , I did not see your new comments when I posted, but yes, I think it is better to separate out the test functions like you said. Perhaps add a note that they can be re-combined when numpy fixes the problem? Thanks!

Reply:
Or maybe what Eero posted?

Reply:
Tried splitting, but @eerovaher's way is nicer, thanks!!